src/utilities/createRouter.ts

```diff
@@ -41,38 +41,35 @@ export function createRouter<const T extends Routes>(routes: T, options: RouterO
 
     const beforeResponse = await runBeforeRouteHooks({ to, from, hooks })
 
-    const setHistory = (): void => history.update(url, { replace })
-    const setRoute = (): void => updateRoute(to)
-
     switch (beforeResponse.status) {
       // On abort do nothing
       case 'ABORT':
         return
 
       // On push update the history, and push new route, and return
       case 'PUSH':
-        setHistory()
+        history.update(url, { replace })
         push(...beforeResponse.to)
         return
 
       // On reject update the history, the route, and set the rejection type
       case 'REJECT':
-        setHistory()
-        setRoute()
+        history.update(url, { replace })
         setRejection(beforeResponse.type)
         break
 
       // On success update history, set the route, and clear the rejection
       case 'SUCCESS':
-        setHistory()
-        setRoute()
```

Comment:
Guessing this is why tests are failing. You'll still need to set the route on reject and success. 

The setHistory and setRoute methods are here because the actions don't care about the logic of how to do these these. Just that it gets done. These actions should also always do the same thing. By removing these we can end up with subtle differences in functionality where there is not supposed to be. So I'd rather not remove them unless you have a strong reason to do so.